runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -317,8 +317,8 @@ function M.save(diagnostics, bufnr, client_id)
 
     -- Clean up our data when the buffer unloads.
     api.nvim_buf_attach(bufnr, false, {
-      on_detach = function(b)
-        clear_diagnostic_cache(b, client_id)
+      on_detach = function()
```

Comment:
why not just `function(_, bufnr)`?

Reply:
I don't think that's necessary here and the other usage of `on_detach` (in lsp.lua) doesn't do that, so I think it's fine to do `function()` instead of `function(_,b)`

Reply:
Not sure about lua, but in other languages creating inline functions that capture scope is more expensive than having inline functions which don't capture variables.

In this case `client_id` is already captured, so I think there isn't a difference between using `function()` and `function(_, b)` here.

The change looks good to me.

Reply:
I still prefer not capturing the scope

Reply:
fixed